os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -113,36 +116,11 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 
 		mm_takesemaphore(heap);
 
-		prev = NULL;
-		node = (struct mm_freenode_s *)heap->mm_heapstart[region];
-		next = (struct mm_freenode_s *)((char *)node + node->size);
-
-		for (; node < (struct mm_freenode_s *)heap->mm_heapend[region]; prev = node, node = next, next = (struct mm_freenode_s *)((char *)next + next->size)) {
-			mvdbg("region=%d node=%p size=%u preceding=%u (%c)\n", region, node, node->size, (node->preceding & ~MM_ALLOC_BIT), (node->preceding & MM_ALLOC_BIT) ? 'A' : 'F');
-
-			if ((prev && prev->size != (node->preceding & ~MM_ALLOC_BIT)) ||
-				(node->size != (next->preceding & ~MM_ALLOC_BIT)) ||
-				(!(node->preceding & MM_ALLOC_BIT) &&
-				((node->blink && node->blink->flink != node) ||
-				(node->flink && node->flink->blink != node)))
-			) {
-				mdbg("ERROR: Heap node corruption detected from one of following nodes\n");
-				if (prev) {
-					mdbg("Previous node addr = 0x%08x size = %u preceding size = %u\n", prev, prev->size, prev->preceding & ~MM_ALLOC_BIT);
-				}
-				mdbg("Current  node addr = 0x%08x size = %u preceding size = %u\n", node, node->size, node->preceding & ~MM_ALLOC_BIT);
-
-#ifdef CONFIG_DEBUG_MM_HEAPINFO
-				if (prev) {
-					mdbg("Previous node owner pid = %u, alloc_call_addr = 0x%08x\n", prev->pid, prev->alloc_call_addr);
-				}
-				mdbg("Current  node owner pid = %u, alloc_call_addr = 0x%08x\n", node->pid, node->alloc_call_addr);
-#endif
-				ASSERT(0);
-			}
+		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; node = (struct mm_allocnode_s *)((char *)node + node->size)) {
+			mvdbg("region=%d node=%p size=%p preceding=%p (%c)\n", region, node, node->size,
```

Comment:
Please print the `node->size` with `%u` instead of `%p`?

Reply:
This was in original code. But you are right. I will change it.

Reply:
I modified the original in https://github.com/Samsung/TizenRT/pull/4791, Anyway Thanks.